lib/mongo/cursor.rb

```diff
@@ -288,7 +288,20 @@ def close
     #
     # @since 2.2.0
     def collection_name
-      @coll_name || collection.name
+      # In most cases, this will be equivalent to the name of the collection
+      # object in the driver. However, in some cases (e.g. when connected
+      # to an Atlas Data Lake), the namespace returned by the find command
+      # may be different, which is why we want to use the collection name based
+      # on the namespace in the command result.
+      if @namespace
+        # Often, the namespace will be in the format "database.collection".
+        # However, sometimes the collection name will contain periods, which
+        # is why this method joins all the namespace components after the first.
+        ns_components = @namespace.split('.')
+        ns_components[1..ns_components.length].join('.')
```

Comment:
`..` includes the end of the range, this needs to either use `...` or be `..ns_components.length-1`.

Reply:
Good catch!